### ISSUE TITLE
ANTs Brain extraction doesn't work for a few subjects in a project

### ISSUE THREAD

**Issue by akelkar26** [ANTsX/ANTs#1392]
**Describe the problem**
I am using ANTs Brain Extraction to extract a skull-stripped brain from the T1w. It doesn't seem to work for 3/35 subjects. These are stroke subjects but I have used this code for another stroke dataset as well and haven't encountered this issue before.

I am using an age-matched template I have created. 
Here is the T1w:
![Capture_t1wtosendtoANTs](https://user-images.githubusercontent.com/40644674/178738987-cfaaebda-5b00-4aaf-aea2-333e53dd5967.PNG)

Here is the extracted brain (which doesn't look like a brain at all):
![Capture_badextractedbrain](https://user-images.githubusercontent.com/40644674/178739434-dd7b09ec-f377-4143-a182-3c2a4191f342.PNG)

**To Reproduce**
Here is the command I used and I am also attaching the data to help reproduce it:
 ${antsscript}ANTsBrainExtraction.sh -d 3 \
        -a $mpr \
        -e $etemplate \
        -m $mtemplate \
        -f $ftemplate \
        -o $bextractdir/ \
        -k 1

T1w attached:

[sub-TMSA24_ses-BRAINRESEARCHCOSLETT2_T1w.nii.gz](https://github.com/ANTsX/ANTs/files/9102427/sub-TMSA24_ses-BRAINRESEARCHCOSLETT2_T1w.nii.gz)

Template files:
etemplate=template.nii.gz
ftemplate=templateBrainExtractionRegistrationMask.nii.gz
mtemplate=templateBrainMaskProbability.nii.gz

[template.nii.gz](https://github.com/ANTsX/ANTs/files/9102531/template.nii.gz)
[templateBrainMaskProbability.nii.gz](https://github.com/ANTsX/ANTs/files/9102534/templateBrainMaskProbability.nii.gz)
[templateBrainExtractionRegistrationMask.nii.gz](https://github.com/ANTsX/ANTs/files/9102537/templateBrainExtractionRegistrationMask.nii.gz)

**System information (please complete the following information)**

 - OS: [Mac OS Big Sur]
 - OS version: [11.6]
 - Type of system: [High computing PC]

**ANTs version information**
 - ANTs code version: ANTs Version: 2.3.5.dev224-g5f1fa
 Compiled: Mar  5 2021 23:24:29
 - ANTs installation type: Compiled from source

Could you please help?
Thank you
Best,
Apoorva

**Comment by ntustison** [comment]
The template and template probability images are 0 bytes.  Did they upload correctly?

**Comment by ntustison** [comment]
There's a variety of sites you could use like wetransfer, dropbox, etc.

**Comment by akelkar26** [comment]
Okay sounds good. I have shared the files with you on this Dropbox link:
https://www.dropbox.com/scl/fo/0o0x38b1a1qtls7v0l8wl/h?dl=0&rlkey=zvkx1i3pvglfy1zuryxzkfcgi

Hope this helps.

**Comment by cookpa** [comment]
It looks like the template has a very different FOV to the subject images. ANTs tries to search for an initial affine transform to compensate, but it can occasionally make a bad choice.

I am running it now without the registration metric mask, which can sometimes improve stability.

Another option would be to compute an initial rigid transform in ITK-SNAP, and then use this as the initial input to antsBrainExtraction.sh (-f).

Another option would be to crop the moving image, using something like this script by Paul Yushkevich.

https://github.com/ftdc-picsl/antsct-aging/blob/master/trim_neck.sh

**Comment by cookpa** [comment]
<img width="1177" alt="image" src="https://user-images.githubusercontent.com/611507/178775128-94f35733-c438-48e4-b06d-05940674c50a.png">

Here's the moving image and template with the registration metric mask. I've used ITK-SNAP's registration tools to align the moving to the fixed center of mass, which is what ANTs does. You can see it's quite a distance from correct.

**Comment by cookpa** [comment]
Looks like removing the metric mask will help - here's the initial transform ANTs found for me

<img width="529" alt="image" src="https://user-images.githubusercontent.com/611507/178776174-2daa6a9b-d0be-4a89-9ea4-d48ff60da112.png">

**Comment by dorianps** [comment]
I think Phil is correct. A long neck in the image is usually a recipe for
registration problems and pipeline failures. Zeroing bottom slices is
usually the first thing we try when we see an image like this failing.

Dorian

**Comment by cookpa** [comment]
<img width="692" alt="image" src="https://user-images.githubusercontent.com/611507/178788388-00a16684-170e-40b8-aa97-e29e0a45b29d.png">

**Comment by akelkar26** [comment]
Hello,

Thank you so much for looking into this issue for me and testing it on the data. 
Yes, I see the template mask and the moving image are not aligned. I will remember to check this if I encounter this issue in any other subjects. 
Thank you for the different troubleshooting tips. I will remember to use them if I find these issues again.
I did see the script for the trimming of the neck when I ran the ACT pipeline for this data.
I will crop the neck if it becomes an issue in any other data in the future.
For now, I was able to replicate the analysis and get good registrations for my 3 subjects with the metric registration mask image not specified. 
This was very helpful and a good learning for any future troubleshooting.
Thank you so much!

Best,
Apoorva
